### ISSUE TITLE
Assertion violation at contractor/contractor_worklist_fixpoint.cc:101 (incomplete fix)

### ISSUE THREAD

**Issue by rainoftime** [dreal/dreal4#200]
Related: https://github.com/dreal/dreal4/issues/199

-----

Hi, for the following formula,
~~~~
(set-logic QF_NRA)
(declare-const r31415926 Real)
(declare-const r31415927 Real)
(assert (<= r31415926 r31415927))
(declare-const r1 Real)
(check-sat)
(check-sat)
(check-sat)
(declare-const r5 Real)
(assert (not (= r1 r5)))
(check-sat)
~~~~

real (commit 215cd00fd) throws an assertion violation
~~~~
delta-sat with delta = 0.001
delta-sat with delta = 0.001
delta-sat with delta = 0.001
dreal: dreal/contractor/contractor_worklist_fixpoint.cc:101: virtual void dreal::ContractorWorklistFixpoint::Prune(dreal::ContractorStatus*) const: Assertion `static_cast<size_t>(branching_point) < input_to_contractors_.size()' failed.
Aborted (core dumped)
~~~~

**Comment by rainoftime** [comment]
Here is another test case
~~~~
(set-logic QF_NRA)
(declare-const r31415926 Real)
(declare-const r31415927 Real)
(assert (< r31415926 r31415927))
(check-sat)
(declare-const r14 Real)
(assert (not (= (- r14) 0.0)))
(check-sat)
~~~~

**Comment by rainoftime** [comment]
Another test case that uses push/pop
~~~~
(set-logic QF_NRA)
(declare-const r31415926 Real)
(declare-const r31415927 Real)
(push 1)
(check-sat)
(pop 1)
(assert (not (< r31415926 r31415927)))
(check-sat)
(declare-const r4 Real)
(push 1)
(assert false)
(check-sat)
(check-sat)
(pop 1)
(assert (or (= (+ r4 23531254.0) 0.0) (not (< r31415926 r31415927))))
(check-sat)
~~~~

**Comment by soonho-tri** [comment]
Fixed by e7cbd70
